drivers/telescope/lx200_10micron.cpp

```diff
@@ -514,6 +550,167 @@ bool LX200_10MICRON::setLocalDate(uint8_t days, uint8_t months, uint16_t years)
     return 0 == setStandardProcedureAndExpect(fd, data, "1");
 }
 
+bool LX200_10MICRON::SetTLEtoFollow(const char *tle)
+{
+    LOGF_INFO("The function is called with TLE %s", tle);
+    if (strlen(tle)>230)
+    {
+        LOG_WARN("TLE is too long");
+    }
+
+    std::string tle_str;
+    std::string sep = "$0a";
+    std::string search = "\n";
+    tle_str = (std::string) tle;
+    for( size_t pos = 0; ; pos += sep.length() ) {
+        // Locate the substring to replace
+        pos = tle_str.find( search, pos );
+        if( pos == std::string::npos ) break;
+        // Replace by erasing and inserting
+        tle_str.erase( pos, search.length() );
+        tle_str.insert( pos, sep );
+    }
+    char command[250];
+    snprintf(command, sizeof(command), ":TLEL0%s#", tle_str.c_str());
+
+    if ( !isSimulation() )
+    {
+        LOG_INFO(command);
+        char response[2];
+        if (0 != setStandardProcedureAndReturnResponse(fd, command, response, 2) )
+        {
+            LOG_ERROR("TLE set error");
+            return 1;
+        }
+        if (response[0] == 'E')
+        {
+            LOG_ERROR("Invalid formatting of TLE, trying to split:");
+            char *pch = strtok ((char*) tle,"\n");
+            while (pch != NULL)
+            {    
+                LOGF_INFO("%s\n",pch);
+                pch = strtok (NULL, "\n");
+            }
+            return 1;
+        }
+    }
+    else
+    {
+        char *pch = strtok ((char*) tle,"\n");
+        while (pch != NULL)
+        { 
+            LOGF_INFO("%s\n",pch);
+            pch = strtok (NULL, "\n");
+        }
+    }
+    return 0;
+}
+	
+bool LX200_10MICRON::SetTLEfromDatabase(int tleN)
+{
+    char command[12];
+    snprintf(command, sizeof(command), ":TLEDL%01d#", tleN);
```

Comment:
question: is %d insufficient here ? does it need to be %01d ? I don't see any hints to that in the docs.

Reply:
Hi, I think %d should be enough